build/common.project.props

```diff
@@ -51,7 +51,7 @@
     <MicroBuildDirectory>$(SolutionPackagesFolder)microsoft.visualstudioeng.microbuild.core\0.4.1\build\</MicroBuildDirectory>
     <MicrosoftDotNetBuildTasksFeedFilePath>$(SolutionPackagesFolder)microsoft.dotnet.build.tasks.feed\6.0.0-beta.20528.5\tools\netcoreapp2.1\Microsoft.DotNet.Build.Tasks.Feed.dll</MicrosoftDotNetBuildTasksFeedFilePath>
     <MicrosoftDotNetMaestroTasksFilePath>$(SolutionPackagesFolder)microsoft.dotnet.maestro.tasks\1.1.0-beta.21117.3\tools\netcoreapp3.1\Microsoft.DotNet.Maestro.Tasks.dll</MicrosoftDotNetMaestroTasksFilePath>
-    <NoWarn>$(NoWarn);NU5105</NoWarn>
+    <NoWarn>$(NoWarn);NU5105;MSB3277</NoWarn>
```

Comment:
Can't we fix the MSB3277 warnings?

Reply:
Originally I thought that MSB3277 was caused by an assembly being referenced by multiple versions of another  assembly, but different versions of that first assembly having being used.  However, looking more closely at the issue, it appears that we might be referencing some 16.x versions of VS components, while the dev17 VSSDK was compiled against higher versions.

So, probably we can resolve them by making sure we use dev17 versions of all our packages. I'll try to make sure I look into it within a few days.

Reply:
@nkolev92 I just spent 2 hours trying to clean up our PackageReferences by removing as many individual VS packages and using the VS SDK everywhere instead.  I found that we're downgrading the threading analyzers in one project, so we need to stop that and fix the warnings that it raises.

However, it still does not resolve the MSB3277 warnings. I believe it's a fundamental challenge with .NET tools in general (msbuild, nuget) when trying to create such a huge SDK, as many of the packages are created independently, and therefore when creating the metapackage, it's impossible to guarantee that the versions of packages/assemblies restored are always higher than the assembly version numbers used in every single compile time reference of the assemblies in the packages.

For example, say the meta package contains packages A and B. A.dll has a compile time reference B.dll version 1.0.1, but package B contains B.dll assembly version 1.0.0.  This will trigger MSB3277. This may be impossible to solve without circular references, hence some packages have undeclared dependencies and therefore minimum versions are not correct in the restore graph.

For now, I propose that we still ignore MSB3277. Migrating (most) projects to the VS SDK instead of individual components should be done in a separate task, as well as fixing as the VS threading warnings that doing this introduces.

Reply:
Sounds good to me.